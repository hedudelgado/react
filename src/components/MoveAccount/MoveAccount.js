import React from 'react';
import { object } from 'prop-types';
import Typography from '../../layout/Typography';
import Hyperlink from '../../layout/Hyperlink';
import { getJsonItem } from '../../utils/session-storage';

const MoveAccount = ({ dictionary }) => (
  <section className="component-container">
    <Typography component="h4" className="pi-color-greyscale-darkest">
      {dictionary['moveaccount.title']}
    </Typography>
    <Typography component="p" mb="3" dangerouslySetInnerHTML={{ __html: dictionary['moveaccount.body'] }} />
    <Hyperlink variant="primary" href={getJsonItem('environment').tetheringUrl}>
      {dictionary['moveaccount.submit']}
    </Hyperlink>
  </section>
);

MoveAccount.propTypes = {
  dictionary: object.isRequired,
};

export default MoveAccount;
