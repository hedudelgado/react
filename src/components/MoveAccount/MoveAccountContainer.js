import { connect } from 'react-redux';
import MoveAccount from './MoveAccount';

const mapStateToProps = ({
  dictionary,
}) => ({
  dictionary,
});

export default connect(mapStateToProps)(MoveAccount);
