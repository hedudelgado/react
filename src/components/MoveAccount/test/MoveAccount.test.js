import React from 'react';
import { shallow } from 'enzyme';
import MoveAccount from '../MoveAccount';

jest.mock('../../../utils/session-storage', () => ({
  __esModule: true,
  getJsonItem: () => ({
    tetheringUrl: 'https://develop--pi-account-tethering.netlify.com',
  }),
}));

describe('MoveAccount Component', () => {
  const props = {
    dictionary: {
      'moveaccount.title': 'Move Account',
      'moveaccount.body': 'A lot of text',
      'moveaccount.submit': 'Move my account',
    },
  };

  const wrapper = shallow(<MoveAccount {...props} />);

  it('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
