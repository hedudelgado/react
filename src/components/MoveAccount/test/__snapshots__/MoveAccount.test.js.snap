// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoveAccount Component should render the component 1`] = `
<section
  className="component-container"
>
  <Typography
    className="pi-color-greyscale-darkest"
    component="h4"
  >
    Move Account
  </Typography>
  <Typography
    component="p"
    dangerouslySetInnerHTML={
      Object {
        "__html": "A lot of text",
      }
    }
    mb="3"
  />
  <Hyperlink
    className=""
    href="https://develop--pi-account-tethering.netlify.com"
    variant="primary"
  >
    Move my account
  </Hyperlink>
</section>
`;
